### DISCUSSION TITLE
Couldn't connect with compose run instead of compose up

### DISCUSSION THREAD

**Discussion by PackmanDude** [itzg/docker-minecraft-server#1619] [Q&A] [answered]
![зображення](https://user-images.githubusercontent.com/99031872/180212017-f740672e-927a-4674-9591-d8cacc790957.png)

(Important thing to look at is a 'PORTS' section)

In the first method I'm using
```bash
$ docker compose run --rm --name server server
```

And in 2nd,
```bash
$ docker compose up --force-recreate -V -d
```

Why can't I connect to my server with the 1st method, because when I use the 2nd method without `--detach` I can't use it interactively, without the need to attach manually after starting. The first method does the same, but interactive tty works as expected. Could someone, please, help me?

**Comment by itzg** [comment, marked as answer]
`docker compose run` is not the command you think it is. It's not applicable for what you're wanting. 

https://docs.docker.com/engine/reference/commandline/compose_run/

**Reply by PackmanDude** [reply]
Thanks, I'm a total noob at Docker. Using `up` won't actually give me a proper shell, because logs are being streamed from all the containers. How can I fix that? I can run it detached and attach it at any moment, but it would be great to be attached from the start (using only 1 command).

**Reply by itzg** [reply]
Can't be done in one command. You'll `docker compose up -d` and then at any time you can 
- `docker compose logs -f` to watch logs
- `docker compose exec SERVICE_NAME bash` to exec a shell inside container
- `docker compose exec SERVICE_NAME rcon-cli` to exec into an rcon session to send minecraft commands to the server

You may want to look through https://docs.docker.com/compose/reference/ and providing your compose file would help if you have any further concerns.

**Reply by PackmanDude** [reply]
Thank you for an explanation!
